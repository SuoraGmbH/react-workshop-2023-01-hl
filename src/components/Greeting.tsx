import React from "react";

interface Props {
  city: string;
}

const Greeting: React.FunctionComponent<Props> = ({ city }) => {
  return <div>Hallo {city.toUpperCase()}!</div>;
};

export default Greeting;
