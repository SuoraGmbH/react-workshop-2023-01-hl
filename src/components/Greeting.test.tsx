import { render, screen } from "@testing-library/react";
import Greeting from "./Greeting";

test("Renders without crashing", () => {
  render(<Greeting city="Hamburg" />);
});

test("It renders the name of the given city", () => {
  render(<Greeting city="Hamburg" />);

  expect(screen.getByText("Hamburg", { exact: false })).toBeInTheDocument();
});

test("It renders the name of the given city in capital letters", () => {
  render(<Greeting city="Hamburg" />);

  expect(screen.getByText(/HAMBURG/)).toBeInTheDocument();
});

test("It renders a greeting", () => {
  render(<Greeting city="Hamburg" />);

  // It's the same as exact: false above.
  expect(screen.getByText(/hallo/i)).toBeInTheDocument();

  // screen.
});

test("It renders very long city names", () => {
  render(
    <Greeting city="Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch" />
  );

  expect(
    screen.getByText(
      "Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch",
      { exact: false }
    )
  ).toBeInTheDocument();
});

test("It renders according to snapshot", () => {
  const { container } = render(
    <Greeting city="Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch" />
  );

  expect(container).toMatchSnapshot();
});

test("Snapshots also work for values", () => {
  expect({
    some: "object 12344",
  }).toMatchInlineSnapshot(`
    Object {
      "some": "object 12344",
    }
  `);

  expect(1 + 2).toMatchInlineSnapshot(`3`);
  expect(1 + 2).toBe(3);
});
