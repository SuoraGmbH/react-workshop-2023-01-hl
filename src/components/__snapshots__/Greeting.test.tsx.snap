// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders according to snapshot 1`] = `
<div>
  <div>
    Hallo 
    LLANFAIRPWLLGWYNGYLLGOGERYCHWYRNDROBWLLLLANTYSILIOGOGOGOCH
    !
  </div>
</div>
`;
